lib/commander/runner.rb

```diff
@@ -237,9 +237,18 @@ def active_command
     ##
     # Attempts to locate a command name from within the arguments.
     # Supports multi-word commands, using the largest possible match.
+    # Returns the default command, if no valid commands found in the args.
 
     def command_name_from_args
-      @__command_name_from_args ||= (valid_command_names_from(*@args.dup).sort.last || @default_command)
+      @__command_name_from_args ||= (longest_valid_command_name_from(@args) || @default_command)
+    end
+
+    ##
+    # Attempts to locate a command name from within the provided arguments.
+    # Supports multi-word commands, using the largest possible match.
+
+    private def longest_valid_command_name_from(args)
+      valid_command_names_from(*args.dup).max
```

Comment:
Extract a new method to keep things DRY.

I've made a small optimisation change here. I predict `max` will be faster than `stort.last` as it'll get the same result without creating a new `Array`.

```diff
- valid_command_names_from(*args.dup).sort.last
+ valid_command_names_from(*args.dup).max
```